subprojects/dependency-management/src/test/groovy/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/ConnectionFailureRepositoryBlacklisterTest.groovy

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.artifacts.ivyservice.ivyresolve
+
+import org.gradle.internal.resource.transport.http.HttpErrorStatusCodeException
+import spock.lang.Specification
+import spock.lang.Subject
+import spock.lang.Unroll
+
+class ConnectionFailureRepositoryBlacklisterTest extends Specification {
+
+    @Subject RepositoryBlacklister blacklister = new ConnectionFailureRepositoryBlacklister()
+
+    def "initializes with no blacklisted repositories"() {
+        expect:
+        blacklister.blacklistedRepositories.empty
```

Comment:
⭕️ This is testing internal implementation details. As far as I can see we are only exposing the `Set` to test it here. I'd remove this and only test through the API we actually care about: `blackList()` and `isBlackListed()`

Reply:
Done.